dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/sequence/FastaSequenceReaderTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.sequence;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.nio.file.Paths;
+import java.util.NoSuchElementException;
+
+import static org.junit.Assert.assertEquals;
+import static uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader.CONTIG_NOT_PRESENT_EXCEPTION_MESSAGE;
+import static uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader.END_LESS_THAN_START_EXCEPTION_MESSAGE;
+import static uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader.QUERY_PAST_END_OF_CONTIG_MESSAGE;
+import static uk.ac.ebi.eva.dbsnpimporter.sequence.FastaSequenceReader.START_NEGATIVE_EXCEPTION_MESSAGE;
+
+public class FastaSequenceReaderTest {
+
+    private FastaSequenceReader reader;
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
`tearDown` to close the reader open before each test?

Reply:
Good idea, I added the close method after implementing those tests.